js/client-battle.js

```diff
@@ -1007,14 +1040,21 @@
 		updateSideLocation: function (sideData) {
 			if (!sideData.id) return;
 			this.side = sideData.id;
-			if (this.battle.sidesSwitched !== !!(this.side === 'p2')) {
+			if (this.battle.sidesSwitched !== !!(this.side === 'p2' || this.side === 'p4')) {
 				this.battle.switchSides();
 				this.$chat = this.$chatFrame.find('.inner');
 			}
 		},
 		updateSide: function () {
 			var sideData = this.request.side;
 			this.battle.myPokemon = sideData.pokemon;
+			this.battle.mySide = this.battle.nearSide = this.battle[sideData.id];
```

Comment:
Err, I don't think `nearSide` should be set to your own side. It should be set to `mySide` or `mySide.ally` depending on which one is `siden <= 1`.

Reply:
So something like this?
```
this.battle.mySide = this.battle[sideData.id];
this.battle.nearSide = this.sides[this.battle.mySide.n % 2];
```

Reply:
Yes! That's what I was thinking.